comp110-worksheet-6/OxoBoard.cs

```diff
@@ -6,71 +6,142 @@
+        // If the square is not empty, leave it as-is and return False.
+        public bool SetSquare(int x, int y, Mark mark)
+        {
+            if (numberMatrix[x, y] == Mark.None)
+            {
+                numberMatrix[x, y] = mark;
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+
+
+            // throw new NotImplementedException("TODO: implement this function and then remove this exception");
+        }
+
+        // If there are still empty squares on the board, return false.
+        // If there are no empty squares, return true.
+        public bool IsBoardFull()
+        {
+            if (numberMatrix[x, y] == Mark.None)
+            {
+                return false;
+            }
+            else
+            {
+                return true;
+            }
+            // throw new NotImplementedException("TODO: implement this function and then remove this exception");
+        }
+
+        // If a player has three in a row, return Mark.O or Mark.X depending on which player.
+        // Otherwise, return Mark.None.
+        public Mark GetWinner()
+        {
+            for (int x = 0; x < 3; x++)
+            {
+                for (int y = 0; y < 3; y++)
+                {
+                    if (numberMatrix[x, 0] == Mark.X && numberMatrix[x, 1] == Mark.X && numberMatrix[x, 2] == Mark.X)
+                    {
+                        return Mark.X;
+                    }
+                    else if (numberMatrix[0, y] == Mark.X && numberMatrix[1, y] == Mark.X && numberMatrix[2, y] == Mark.X)
+                    {
+                        return Mark.X;
+                    }
+                    else if (numberMatrix[x, 0] == Mark.O && numberMatrix[x, 1] == Mark.O && numberMatrix[x, 2] == Mark.O)
+                    {
+                        return Mark.O;
+                    }
+                    else if (numberMatrix[0, y] == Mark.O && numberMatrix[1, y] == Mark.O && numberMatrix[2, y] == Mark.O)
+                    {
+                        return Mark.O;
+                    }    
+                }         
+            }
+            if (numberMatrix[0, 0] == Mark.X && numberMatrix[1, 1] == Mark.X && numberMatrix[2, 2] == Mark.X)
+            {
+                return Mark.X;
+            }
+            else if (numberMatrix[0, 2] == Mark.X && numberMatrix[1, 1] == Mark.X && numberMatrix[2, 0] == Mark.X)
+            {
+                return Mark.X;
+            }
+            else if (numberMatrix[0, 0] == Mark.O && numberMatrix[1, 1] == Mark.O && numberMatrix[2, 2] == Mark.O)
+            {
+                return Mark.O;
+            }
+            else if (numberMatrix[0, 2] == Mark.O && numberMatrix[1, 1] == Mark.O && numberMatrix[2, 0] == Mark.O)
+            {
+                return Mark.O;
+            }
+            // throw new NotImplementedException("TODO: implement this function and then remove this exception");
```

Comment:
This gives an error "not all code paths return a value" -- you need to `return Mark.None` in the case that no line was detected